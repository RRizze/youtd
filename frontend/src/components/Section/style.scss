.section {
  padding-top: 35px;
  background-color: #2b2533;
  color: #cadce8;
}