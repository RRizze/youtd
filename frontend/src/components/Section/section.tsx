import React, { ReactNode } from "react";

import "./style.scss";

interface Props {
  caption?: string;
  className: string;
  children: ReactNode;
}

const Section = ({
  caption,
  className,
  children,
}: Props) => (
    <section className={`section ${className}`}>
      <div className="wrap">
        {caption &&
          <h1>{caption}</h1>
        }
        {children}
      </div>
    </section>
  );

export default Section;
