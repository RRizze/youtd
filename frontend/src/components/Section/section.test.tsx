import React from "react";
import { mount } from "enzyme";
import Section from "./section";
import renderer from "react-test-renderer";

describe("Section tests", () => {
  it("correct render Section", () => {
    const props = {
      caption: null,
      className: null,
      children: null,
    };
    const tree = renderer
      .create(<Section {...props}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it("render with empty props", () => {
    const props = {
      caption: null,
      className: null,
      children: null,
    },
    SectionComponent = mount(<Section {...props} />);
    expect(SectionComponent.props()).toEqual(props);
  });

  it("render with props", () => {
    const props = {
      caption: "About",
      className: "about",
      children: "andufalador",
    },
    SectionComponent = mount(<Section {...props} />);
    expect((SectionComponent).props()).toEqual(props);
    expect(SectionComponent.find("h1").html()).toMatch("About");
  });
});
