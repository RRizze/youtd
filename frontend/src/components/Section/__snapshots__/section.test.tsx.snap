// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section tests correct render Section 1`] = `
<section
  className="section null"
>
  <div
    className="wrap"
  />
</section>
`;
